crates/ruff_python_parser/src/lib.rs

```diff
@@ -610,6 +655,36 @@ impl Deref for Tokens {
     }
 }
 
+/// One challenge here is how to we keep the spine to the root.
```

Comment:
I don't really know the parser code so feel free to ignore, but I don't really understand this comment. Tokens are a stream, not a tree, so what does "spine to the root" even mean in terms of tokens?